index.bs

```diff
@@ -387,233 +507,172 @@ Each \[[Stage]] value indicates the currently pending operation. If the \[[Resul
+1. If _stage_ is "ready" or *undefined*, then:
+  1. Return the result of transforming RequestReady(_entry_) with a fulfillment handler that, when called with argument _entry_, runs the following steps:
+    1. Return GetModuleNamespace(_entry_.[[Module]]).
+1. Return a promise rejected with a new *RangeError* exception.
 </emu-alg>
 
-<h4 id="registry-prototype-cancel">Registry.prototype.cancel(key)</h4>
+<h4 id="registry-entry-prototype-result">RegistryEntry.prototype.result(stage)</h4>
 
 The following steps are taken:
 
 <emu-alg>
-1. Let _registry_ be *this* value.
-1. If Type(_registry_) is not Object, throw a *TypeError* exception.
-1. Let _entries_ be _registry_.[[RegistryData]].
-1. Let _pair_ be the entry in _entries_ such that _pair_.[[key]] is equal to _key_.
-1. If _pair_ does not exist, then throw a new *TypeError*.
-1. Let _entry_ be _pair_.[[value]].
-1. Let _stageEntry_ be GetCurrentStage(_entry_).
-1. If _stageEntry_.[[Stage]] is "link" or "ready", throw a new *TypeError*.
-1. Remove _pair_ from _entries_.
+1. Let _entry_ be *this* value.
+1. If Type(_entry_) is not Object, return a promise rejected with a new *TypeError* exception.
+1. If _stage_ is not "fetch", "translate", "instantiate", "satisfy", "link" or "ready", return a promise rejected with a new *RangeError* exception.
+1. Let _stageEntry_ be GetStage(_entry_, _stage_).
+1. If _stageEntry_ is an abrupt completion, return a promise rejected with a new *TypeError* exception.
+1. If _stageEntry_ is *undefined*, return a promise resolved with *undefined*.
+1. If _stageEntry_.[[Result]] is *undefined*, return a promise resolved with *undefined*.
+1. Return the result of transforming _stageEntry_.[[Result]] with a new pass-through promise.
 </emu-alg>
 
-<h4 id="registry-prototype-provide">Registry.prototype.provide(key, stage, value)</h4>
+<h4 id="registry-entry-prototype-provide">RegistryEntry.prototype.resolve(stage, result)</h4>
 
 The following steps are taken:
 
 <emu-alg>
-1. Let _registry_ be *this* value.
-1. If Type(_registry_) is not Object, throw a new *TypeError*.
-1. If _stage_ is not "fetch", "translate" or "instantiate", throw a new *TypeError*.
-1. Let _loader_ be _registry_.[[Loader]].
-1. Let _entry_ be EnsureRegistered(_loader_, _key_).
-1. Let _stageEntry_ be GetStage(_entry_, _stage_).
-1. If _stageEntry_ is *undefined*, throw a new *TypeError*.
-1. If _stageEntry_.[[Result]] is *undefined*, then set _entry_.[[Result]] to a promise resolved with _value_.
-1. Else, fulfill _entry_.[[Result]] with _value_.
-1. UpgradeToStage(_entry_, _stage_).
+1. Let _entry_ be *this* value.
+1. If Type(_entry_) is not Object, throw a new *TypeError*.
+1. If _stage_ is not "fetch", "translate", "instantiate", "satisfy", "link" or "ready", return a promise rejected with a new *RangeError* exception.
+1. Let _p0_ be the result of transforming _result_ with a new pass-through promise.
```

Comment:
I have been using two forms:

1. `Let _p0_ be the result of transforming _result_ with a new pass-through promise.`
2. `Let _p0_ be a promise resolved with _result_.`

The main difference here is that sometimes we want to store resolve and reject in some internal slots so we can force to resolve those (at least in the past we were forcing those), this is mostly when we pass the promise around, in the other hand, if the result promise is never exposed, we normally use the second option, which is just Promise.resolve().